nilearn/decoding/decoder.py

```diff
@@ -1464,20 +1500,33 @@ def __init__(
         )
 
     def _more_tags(self):
+        """Return estimator tags.
+
+        TODO remove when bumping sklearn_version > 1.5
+        """
         return self.__sklearn_tags__()
 
     def __sklearn_tags__(self):
+        """Return estimator tags.
+
+        See the sklearn documentation for more details on tags
+        https://scikit-learn.org/1.6/developers/develop.html#estimator-tags
+        """
         # TODO
         # get rid of if block
         # bumping sklearn_version > 1.5
         # see https://github.com/scikit-learn/scikit-learn/pull/29677
-        ver = parse(sklearn_version)
-        if ver.release[1] < 6:
-            return {"multioutput": True}
+        if SKLEARN_LT_1_6:
+            from nilearn._utils.tags import tags
+
+            return tags(multioutput=True, niimg_like=True, surf_img=True)
+
+        from nilearn._utils.tags import InputTags
+
         tags = super().__sklearn_tags__()
-        if ver.release[1] >= 6:
-            tags.estimator_type = "regressor"
+        tags.estimator_type = "regressor"
         tags.target_tags.required = True
+        tags.input_tags = InputTags(niimg_like=True, surf_img=True)
```

Comment:
Same question for the DecoderRegressor?

Reply:
Also I am realizing that we haven't updated the docstring for DecoderRegressor about support for Surface mask/images etc. For another PR...